### PR TITLE
Fixes for recent ovs bridge related changes

### PR DESCRIPTION
**Pull request by rhafer**

### CODE REVIEW

File: chef/cookbooks/network/recipes/default.rb [hunk 1]
```diff
@@ -509,4 +511,18 @@ def kill_nic(nic)
...
+  # should be correctly configure at this point.
+  template "/etc/wicked/local.conf" do
+    source "wicked-local.conf.erb"
+    owner "root"
+    group "root"
+    mode "0644"
+    variables(
+      use_nanny: ovs_setup_once
+    )
+  end
```
Comment (rhafer) [inline chef/cookbooks/network/recipes/default.rb]: Hm, I think I am missing an ```only_if``` condition here. And using ```ovs_setup_once``` once for ```use_nanny``` might also be wrong. I'll reconsider.

File: chef/cookbooks/network/recipes/default.rb [hunk 2]
```diff
@@ -509,4 +511,18 @@ def kill_nic(nic)
...
+  # for the next boot.
+  # Note: There's no need to restart wicked here as all interfaces
+  # should be correctly configure at this point.
+  template "/etc/wicked/local.conf" do
+    source "wicked-local.conf.erb"
+    owner "root"
+    group "root"
+    mode "0644"
+    variables(
+      use_nanny: ovs_setup_once
```
Comment (vuntz) [inline chef/cookbooks/network/recipes/default.rb]: Just checking: there's no plan to restart wicked when that file changes, or is there? That was breaking HA when we had that.

Comment (rhafer) [inline chef/cookbooks/network/recipes/default.rb]: No, see the comment above the template resource. :)

Comment (vuntz) [inline chef/cookbooks/network/recipes/default.rb]: Heh, sorry, was reading too quickly :-)

**Comment by rhafer** [comment]
Reworked. Should be ready now. Removed the wip label.

**Comment by vuntz** [comment]
+1

**Comment by rsalevsky** [comment]
+1

**Comment by rhafer** [comment]
rebased

File: chef/cookbooks/network/recipes/default.rb [hunk 3]
```diff
@@ -509,4 +511,26 @@ def kill_nic(nic)
...
+    # Note: There's no need to restart wicked here as all interfaces
+    # should be correctly configure at this point.
+    template "/etc/wicked/local.conf" do
+      source "wicked-local.conf.erb"
+      owner "root"
+      group "root"
+      mode "0644"
+      variables(
+        use_nanny: true
+      )
```
Comment (vuntz) [inline chef/cookbooks/network/recipes/default.rb]: Hrm, can you put the whole code inside a big if `node[:platform_family] == "suse" && !(node[:platform] == "suse" && node[:platform_version] < 12.0)` ?

Comment (rhafer) [inline chef/cookbooks/network/recipes/default.rb]: @vuntz Hm, we are already in the  ```when "suse"``` branch of ```case node["platform"]``` here. So I guess only the ```if !node[:platform_version] < 12.0``` would make sense here, right? Or probably better: ```node[:platform_version] >= 12.0```

Comment (vuntz) [inline chef/cookbooks/network/recipes/default.rb]: Ah, right. So the `case` will change to `case node[:platform_family]` soon (see https://github.com/crowbar/crowbar-core/pull/53), so the test should become `unless node[:platform] == "suse" && node[:platform_version] < 12.0` (to make it really SLE11-specific, and not considering openSUSE and others). Technically, just looking at the version would work, but I prefer to be slightly more explicit.

Comment (rhafer) [inline chef/cookbooks/network/recipes/default.rb]: Hm, that code is not SLES11 specific. It's for everything that has wicked. (SLE12 and newer, including openSUSE if we supported that)

Comment (vuntz) [inline chef/cookbooks/network/recipes/default.rb]: Hence the `unless` and not `if` :-)

**Comment by vuntz** [comment]
Gating doesn't pass because the test package can't be found anymore; need to rebase or manually trigger a package rebuild.

File: chef/cookbooks/network/recipes/default.rb (hunk 3 above)
Comment (rhafer) [inline chef/cookbooks/network/recipes/default.rb]: Addressed.

**Comment by vuntz** [comment]
+1

**Comment by mjura** [comment]
+1 looks OK

I'm wondering if this time gating will finish successfully

**Comment by rhafer** [comment]
> I'm wondering if this time gating will finish successfully

Me too ;). Though none of the failures was related to this PR itself :/

**Comment by vuntz** [comment]
You're reaching cct, so tempest is passing => good enough for merging.

**Pull request merged**